libkineto/ipcfabric/Endpoint.h

```diff
@@ -163,8 +174,21 @@ class EndPoint final {
     throw std::runtime_error(std::strerror(errno));
   }
 
-  const char* getName(TCtxt const& ctxt) const noexcept {
-    return ctxt.msg_name.sun_path + 1;
+  const char* getName(TCtxt const& ctxt) const {
+    const char* socket_dir = getenv("KINETO_IPC_SOCKET_DIR");
+    bool is_domain_socket = socket_dir && socket_dir[0];
+    if (is_domain_socket) {
+      int prefix_len = strlen(socket_dir) + 1;
+      if (strncmp(socket_dir, ctxt.msg_name.sun_path, prefix_len) == 0) {
```

Comment:
@dzhulgakov is this meant to be '!=' 0 instead of == 0

Reply:
It should be, my bad! It didn't trigger because I put `prefix_len` to be +1. C strings are hard :)